src/__tests__/FileSynchronizer.test.ts

```diff
@@ -0,0 +1,133 @@
+import { AbortController } from 'abortcontroller-polyfill/dist/cjs-ponyfill';
+
+import { FileSynchronizer } from '../FileSynchronizer';
+import { FileInfo, SyncOptions } from '../types';
+
+const defaultOptions = {
+  root: 'test-utils',
+};
+
+async function stub(options: SyncOptions) {
+  const sync = new FileSynchronizer(options);
+
+  const files: FileInfo[] = [];
+  const excludedFiles: FileInfo[] = [];
+
+  sync.on('file', (fileInfo) => {
+    files.push(fileInfo);
+  });
+
+  sync.on('excluded-file', (fileInfo) => {
+    excludedFiles.push(fileInfo);
+  });
+
+  sync.on('end', () => {});
+
+  const controller = new AbortController();
```

Comment:
no need to create AbortController unless it's a test that calls `controller.abort()`